src/DotNetLightning.Core/Channel/ChannelHelpers.fs

```diff
@@ -17,3 +24,431 @@ module internal ChannelConstantHelpers =
+                    (staticChannelConfig.RemoteParams.DustLimitSatoshis)
+                    localCommitmentPubKeys.RevocationPubKey
+                    staticChannelConfig.LocalParams.ToSelfDelay
+                    remoteCommitmentPubKeys.DelayedPaymentPubKey
+                    localCommitmentPubKeys.PaymentPubKey
+                    remoteCommitmentPubKeys.HtlcPubKey
+                    localCommitmentPubKeys.HtlcPubKey
+                    remoteCommit.Spec
+                    staticChannelConfig.Network
+
+            ClaimMainOutput
+                (commitTx.Value.GetGlobalTransaction())
+                remoteCommit.Index
+                staticChannelConfig
+                (Choice1Of2 remotePerCommitmentSecret)
+                localChannelPrivKeys
+
+        let ClaimCommitTxOutputs
+            (closingTx: Transaction)
+            (staticChannelConfig: StaticChannelConfig)
+            (remotePerCommitmentSecrets: PerCommitmentSecretStore)
+            (channelPrivKeys: ChannelPrivKeys)
+            =
+            let obscuredCommitmentNumberRes =
+                tryGetObscuredCommitmentNumber
+                    staticChannelConfig.FundingScriptCoin.Outpoint
+                    closingTx
+
+            match obscuredCommitmentNumberRes with
+            | Ok obscuredCommitmentNumber ->
+                let localChannelPubKeys = channelPrivKeys.ToChannelPubKeys()
+
+                let remoteChannelPubKeys =
+                    staticChannelConfig.RemoteChannelPubKeys
+
+                let commitmentNumber =
+                    obscuredCommitmentNumber.Unobscure
+                        staticChannelConfig.IsFunder
+                        localChannelPubKeys.PaymentBasepoint
+                        remoteChannelPubKeys.PaymentBasepoint
+
+                let claimMainOutputsRes =
+                    ClaimMainOutput
+                        closingTx
+                        commitmentNumber
+                        staticChannelConfig
+                        (Choice2Of2 remotePerCommitmentSecrets)
+                        channelPrivKeys
+
+                match claimMainOutputsRes with
+                | Error OutputClaimError.UnknownClosingTx ->
+                    {
+                        MainOutput = Error OutputClaimError.UnknownClosingTx
+                    }
+                | claimMainOutput ->
+                    {
+                        MainOutput = claimMainOutput
+                    }
+            | _ ->
+                {
+                    MainOutput = Error UnknownClosingTx
+                }
+
+    let HandleFundingTxSpent
+        (savedChannelState: SavedChannelState)
+        (remoteNextCommitInfoOpt: Option<RemoteNextCommitInfo>)
+        (channelPrivKeys: ChannelPrivKeys)
+        (closingTx: Transaction) : ClosingResult =
+        let closingTxId = closingTx.GetTxId()
+        if closingTxId = savedChannelState.LocalCommit.PublishableTxs.CommitTx.Value.GetTxId() then
```

Comment:
Is there any other way to recognize local commitments? (This current check only works for last local commitment and not previous/revoked ones)

Reply:
🤷